be/src/storage/meta_reader.cpp

```diff
@@ -160,8 +160,8 @@ Status SegmentMetaCollecter::_init_return_column_iterators() {
         if (_params->read_page[i]) {
             auto cid = _params->cids[i];
             if (_column_iterators[cid] == nullptr) {
-                ASSIGN_OR_RETURN(_column_iterators[cid],
-                                 _segment->new_column_iterator(cid, nullptr, _params->tablet_schema));
+                const TabletColumn& col = _params->tablet_schema->column(cid);
```

Comment:
Are you sure that `_params->tablet_schema` will be always not null?

Reply:
Yes I think so, I've investigated the call stack for this function and the `_params->tablet_schema` here should not be null.

Reply:
Your concerns are valid though, and more checks should be added here to ensure the robustness of the program.

Reply:
Erm, you did find a bug, the `_params->tablet_schema` here would indeed be nullptr in shared data mode. Since our CI is not running in shared data mode, this issue was not discovered.

Reply:
Fixed, PTAL @imay